File: python/busca_ordenada.py
def burble_sort(vetor):
    for i in range(len(vetor)):
        for i in range(len(vetor) - 1):
            if vetor[i] > vetor[i+1]:
                auxiliar = vetor[i]
                vetor[i] = vetor[i+1]
                vetor[i+1] = auxiliar
    return vetor
    
def busca_sequecial(numero, vetor):
    for i in range(len(vetor)):
        if numero == vetor[i]:
            print(f"O número {numero} está na posição {i} do vetor: {vetor}\nOnde o vetor se inicia com o número {vetor[0]} estando na posição 0")
            break
    else :
        print("Número não encontrado.")
        
def add_vetor(adicionar, vetor):
    
    if adicionar == 1:
        novo_num = int(input(f"Digite o número que deseja adicionar no vetor {vetor}: "))
        vetor.append(novo_num)
        burble_sort(vetor)
        encontrar = int(input(f"Digite o número que deseja encontrar na vetor {vetor}: "))
        busca_sequecial(encontrar, vetor)
    else:
        print("Obrigado por testar o código!")
        
    
vetor = [5, 1, 39, 28, 104, 90, 80, -39]
#vetor.sort()
#função pronta do python para ordenar os elementos do vetor
burble_sort(vetor)

numero = int(input(f"Digite o número que deseja encontrar na vetor {vetor}: "))
busca_sequecial(numero, vetor)

adicionar = int(input('Deseja adicionar mais números ao vetor? [digite 1 para "sim" ou 0 para "não"] '))
add_vetor(adicionar, vetor)

File: C++/busca_ordenada.cpp
#include <iostream>

void burble_sort(int vetor[], int tamanho){
    int auxiliar;
    for(int j = 0; j < tamanho; j++){
        for(int i = 0; i < tamanho - 1; i++){
            if(vetor[i] > vetor[i+1]){
                auxiliar = vetor[i];
                vetor[i] = vetor[i+1];
                vetor[i+1] = auxiliar;
            }
            
        }
    }
}

void busca_sequencial(int valor, int vetor[], int tamanho){
    bool encontrado = false;
    if(valor == vetor[0]){
        std::cout<< "O valor está na posição 0."<< std::endl;
        encontrado = true;
    }
    for(int i = 1; i <= tamanho; i++){
        if(valor == vetor[i]){
            std::cout<< "O valor está na posição " << i << ", onde o valor " << vetor[0] << " se encontra na posição 0."<< std::endl;
            encontrado = true;
        }
    }
    if(!encontrado){
        std::cout << "O valor " << valor << " não está no vetor." << std:: endl;
    }
}

int main(){
    int valor, vetor[] = {30, 28, 83, 14, 56, 356, 97, 108, 9999, 200}, tamanho = sizeof(vetor) / sizeof(vetor[0]);
    
    burble_sort(vetor, tamanho);


    std::cout << "Escolha o número que deseja buscar no vetor [";
    for(int i = 0; i < tamanho; i++){
        std::cout << vetor[i];
        if(i < tamanho - 1){
            std::cout << ", ";
        }
    }
    std:: cout << "]: ";
    std::cin >> valor;

    busca_sequencial(valor, vetor, tamanho);    

    return 0;
}